cibyl/models/ci/system.py

```diff
@@ -95,8 +95,7 @@ class System(Model):
         this system. For most system's, this will be :class:`Job`.
     """
     API = {
-        **BASE_SYSTEM_API,
-        **JOBS_SYSTEM_API
+        **BASE_SYSTEM_API
     }
```

Comment:
I spotted a bug here where if a Zuul and jobs system where used at the same time, the jobs system API would be lost. I do not have test cases for it, I just noticed by quick look.

Reply:
I think that the bug comes from using the unpacking operator, that creates a shallow copy. Also, wouldn't this implementation have a sort of "race condition" where the final API would be the one of the last system created? I think the simplest solution would be to keep the System API as it was, set the JobsSystem API using a deepcopy of the System one, and creating a different API object for ZuulSystem (manually, without reusing the BASE_SYSTEM_API) to avoid the same issue

Reply:
I have added a test: test_aggregated_system_api, to prove that the API is built as expected. Every time a new kind of system is created, they install their subset of parameters into System.API. It is just simple addition.

Reply:
Right, however it should be noted that this approach constructs a slightly different hierarchy than before when both systems are in use. Since now jobs can be either at the same level as tenant, or below, depending on which of the two is first read when extending the parser. I think that this is ok for now, but in a separate task we should move the code that modifies the System API out of the constructors of the systems and into a separate method of the orchestrator, after the environments are created, for example.